File: monolith-api/src/main/scala/gg/warcraft/monolith/api/world/block/BlockColor.scala
package gg.warcraft.monolith.api.world.block

import enumeratum._

import scala.collection.immutable.IndexedSeq

sealed trait BlockColor extends EnumEntry

object BlockColor extends Enum[BlockColor] {
  val values: IndexedSeq[BlockColor] = findValues

  case object BLACK extends BlockColor
  case object BLUE extends BlockColor
  case object BROWN extends BlockColor
  case object CYAN extends BlockColor
  case object GRAY extends BlockColor
  case object GREEN extends BlockColor
  case object LIGHT_BLUE extends BlockColor
  case object LIGHT_GRAY extends BlockColor
  case object LIME extends BlockColor
  case object MAGENTA extends BlockColor
  case object ORANGE extends BlockColor
  case object PINK extends BlockColor
  case object PURPLE extends BlockColor
  case object RED extends BlockColor
  case object WHITE extends BlockColor
  case object YELLOW extends BlockColor
}

File: monolith-api/src/main/scala/gg/warcraft/monolith/api/world/block/BlockColor.java
package gg.warcraft.monolith.api.world.block;

public enum BlockColor {
    BLACK("Black"),
    BLUE("Blue"),
    BROWN("Brown"),
    CYAN("Cyan"),
    GRAY("Gray"),
    GREEN("Green"),
    LIGHT_BLUE("Light Blue"),
    LIGHT_GRAY("Light Gray"),
    LIME("Lime"),
    MAGENTA("Magenta"),
    ORANGE("Orange"),
    PINK("Pink"),
    PURPLE("Purple"),
    RED("Red"),
    WHITE("White"),
    YELLOW("Yellow");

    public final String capitalized;

    BlockColor(String capitalized) {
        this.capitalized = capitalized;
    }
}
